hoomd_polymers/base/molecule.py

```diff
@@ -0,0 +1,328 @@
+
+    @property
+    def pair_types(self):
+        return self.topology_information["pair_types"]
+
+    @property
+    def bond_types(self):
+        return self.topology_information["bond_types"]
+
+    @property
+    def angle_types(self):
+        return self.topology_information["angle_types"]
+
+    @property
+    def diherdal_types(self):
+        return self.topology_information["dihedral_types"]
+
+    @property
+    def improper_types(self):
+        return self.improper_types["improper_types"]
+    
+    def coarse_grain(self, beads=None, mapping=None):
+        for comp in self.molecules:
+            cg_comp = CG_Compound(comp, beads=beads, mapping=mapping)
+            self._cg_molecules.append(cg_comp)
+
+    def _load(self):
+        if self.file and isinstance(self.file, str): # Loading from file takes precedent over SMILES 
+            return mb.load(self.file)
+        elif self.smiles and isinstance(self.smiles, str):
+            return mb.load(self.smiles, smiles=True)
+        else:
+            raise ValueError(
+                    "Unable to load from ",
+                    f"File: {self.file}",
+                    f"SMILES: {self.smiles}"
+            )
+
+    def _generate(self):
+        for i in range(self.n_mols):
+            self._molecules.append(self._mb_molecule)
+
+    def _convert_to_gmso(self):
+        topology = from_mbuild(self._mb_molecule)
+        topology.identify_connections()
+        return topology
+
+    def _particle_information(self, gmso_molecule):
+        particle_types = set()
+        hydrogen_types = set()
+        for site in gmso_molecule.sites:
+            particle_types.add(site.atom_type or site.name)
+            if site.element.atomic_number == 1:
+                hydrogen_types.add(site.atom_type or site.name)
+        return particle_types, hydrogen_types
+
+    def _identify_pairs(self, particle_types):
+        pairs = list(itertools.combinations_with_replacement(particle_types, 2))
+        return pairs
+
+    def _identify_bond_types(self, gmso_molecule):
+        bond_types = set()
+        for bond in gmso_molecule.bonds:
+            bond_connections = [bond.connection_members[0].atom_type or bond.connection_members[0].name,
+                            bond.connection_members[1].atom_type or bond.connection_members[1].name]
+            bond_connections.sort()
+            bond_types.add(tuple(bond_connections))
+        return bond_types
+
+    def _identify_angle_types(self, gmso_molecule):
+        angle_types = set()
+        for angle in gmso_molecule.angles:
+            angle_connections = [angle.connection_members[0].atom_type or angle.connection_members[0].name,
+                                 angle.connection_members[1].atom_type or angle.connection_members[1].name,
+                                 angle.connection_members[2].atom_type or angle.connection_members[2].name]
+            angle_connections.sort()
```

Comment:
The order of the angles is important, so I don't think we should use sort in this way.  For example, a C-S-C angle is different than a C-C-S angle, and a molecule might have both of them.

Reply:
@marjanAlbouye I don't think we need to worry too much about doing any sorting here. It's important that everything is sorted correctly when creating the hoomd force objects, but gmso does that in the `convert_hoomd` stuff.  Just having non-sorted types of angles and dihedrals present is probably good enough for what we're trying to do here in just providing a quick summary of the types of BADs.